tests/baselines/reference/intraExpressionInferencesJsx.errors.txt

```diff
@@ -0,0 +1,115 @@
+tests/cases/conformance/types/typeRelationships/typeInference/intraExpressionInferencesJsx.tsx(107,17): error TS18046: 'arg' is of type 'unknown'.
+
+
+==== tests/cases/conformance/types/typeRelationships/typeInference/intraExpressionInferencesJsx.tsx (1 errors) ====
+    /// <reference path="/.lib/react16.d.ts" />
+    
+    // repro from #52798
+    
+    type A = {
+      a: boolean;
+    };
+    
+    type B = {
+      b: string;
+    };
+    
+    type C = {
+      c: number;
+    };
+    
+    type Animations = {
+      [key: string]: { value: number } & (
+        | ({ kind: "a"; func?(): Partial<A> } & A)
+        | ({ kind: "b"; func?(): Partial<B> } & B)
+        | ({ kind: "c"; func?(): Partial<C> } & C)
+      );
+    };
+    
+    type StyleParam<T extends Animations> = Record<keyof T, string>;
+    
+    type AnimatedViewProps<T extends Animations> = {
+      style: (animationsValues: StyleParam<T>) => string;
+      animations: T;
+    };
+    
+    const Component = <T extends Animations>({
+      animations,
+      style,
+    }: AnimatedViewProps<T>) => <></>;
+    
+    <Component
+      animations={{
+        test: {
+          kind: "a",
+          value: 1,
+          a: true,
+        },
+      }}
+      style={(anim) => {
+        return "";
+      }}
+    />;
+    <Component
+      animations={{
+        test: {
+          kind: "a",
+          value: 1,
+          a: true,
+          func() {
+            return {
+              a: true,
+            };
+          },
+        },
+      }}
+      style={(anim) => {
+        return "";
+      }}
+    />;
+    <Component
+      animations={{
+        test: {
+          kind: "a",
+          value: 1,
+          a: true,
+          func: () => {
+            return {
+              a: true,
+            };
+          },
+        },
+      }}
+      style={(anim) => {
+        return "";
+      }}
+    />;
+    
+    // repro from #52786
+    
+    interface Props<T> {
+      a: (x: string) => T;
+      b: (arg: T) => void;
+    }
+    
+    function Foo<T>(props: Props<T>) {
+      return <div />;
+    }
+    
+    <Foo
+      a={() => 10}
+      b={(arg) => { arg.toString(); }}
+    />;
+    
+    <Foo
+      a={(x) => 10}
+      b={(arg) => { arg.toString(); }}
+    />;
+    
+    <Foo {...{
+      a: (x) => 10,
+      b: (arg) => { arg.toString(); },
+                    ~~~
+!!! error TS18046: 'arg' is of type 'unknown'.
+    }} />;

```

Comment:
From what I see this is the only "unwanted" change that has been found on `main`. However... this is exactly how regular call expressions work: [TS playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgApQPYAcDOAeAFQD5kBvAKGWTgC5kAKADzpzClAHMBKZAXhIIBuSsgBGdenCgc6BHv2QA3DMAAmwgL7lyMAK4gEYYBhDIAYhgyEi9LJlx102fMR4UqUCGF1RTeVcCKyAD0RJraeBYYIqQAdPHuVNQSjPIkAIwADAA0IlTiDFLcfCSJSdTSsWAYAMpsnPRcwuUauVQaWqHC5FH0ifGxZckMqSXIWW1JBZLSaWR5SUVVtfUgHI3NSa0i2xpN5EA). So perhaps... this was just working accidentally before. I feel like this should be a separate issue to make it work for both JSX and regular calls (if that's desired)